### ISSUE TITLE
Connection between the onboarding world and the main world 

### ISSUE THREAD

**Issue by ninadu07** [facebookresearch/Mephisto#563]
During local development, after turkers complete the onboarding tasks and world_data is submitted, the two turkers are connected into the main world for the task. However, after around 15 seconds, the right-bottom response input pane always changes to 
`<div id="response-type-done" class="response-type-module" style="padding: 20px 25px; float: left;"></div> `
and the input pane is hidden/removed from the chat pane.  

This happens every time the turkers complete onboard tasks (with a new onboarding_qualification string) then connect to the main world. With the same onboarding_qualification, if the turkers have gotten onboard previously, then they go straight to the main world and the input pane works fine. 

Any ideas what am I missing here?  Thanks in advance!

**Comment by JackUrb** [comment]
This definitely sounds like a bug - will investigate, thanks for flagging!

**Comment by kushalchawla** [comment]
Hi. I am facing something similar. I also observe this in the example parlai chat demo script when run with the conf=onboarding_example:

`python parlai_test_script.py conf=onboarding_example`


 The behavior is a bit odd: 

At the end of the onboarding parley, we send a final observe to the agent with episode_done=true ([Line 20 here](https://github.com/facebookresearch/Mephisto/blob/dfc985716ddaf552ff57d179d764bfb5d311edad/examples/parlai_chat_task_demo/demo_worlds.py#L20)). Once that happens, ideally I would expect the message box to become inactive and the control would wait to get connected to the partner. But instead, the message box is active and disappears in ~15 seconds. In the meantime that it does not disappear, if I mess around by sending another message, that message then actually interferes with the upcoming chat that happens when the two partners are paired up.

**Comment by kushalchawla** [comment]
Hello. I did some digging. 

This is probably due to [Line 190-209 here](https://github.com/facebookresearch/Mephisto/blob/dfc985716ddaf552ff57d179d764bfb5d311edad/mephisto/abstractions/blueprints/parlai_chat/parlai_chat_task_runner.py#L190).

After calling parley, there is one more round of observe and act. I am not sure why this is the case, even for the onboarding session.

Another thing to note is that the check `hasattr(world, "prep_save_data")` would probably not make a difference since `prep_save_data()` is defined in `CrowdDataWorld`, from which all the onboarding and task worlds inherit. See [here](https://github.com/facebookresearch/ParlAI/blob/c14b3e70a18b9e0ea4ac5fd43428a5c8ee98dd4c/parlai/crowdsourcing/utils/worlds.py#L11).

**Comment by JackUrb** [comment]
Indeed we can drop the `prep_save_data` check now, but this is overall a good lead. Onboarding should be receiving a submission packet here that closes the world, but something is amiss. I should get to resolve later today.

**Comment by JackUrb** [comment]
Alright this should be resolved in #573, thanks for identifying and helping with the debugging! Feel free to reopen this if you are running into more issues (after upgrading your `mephisto-task` and `bootstrap-chat` packages in your local projects, if necessary, as well).

**Comment by kushalchawla** [comment]
Hry @JackUrb 

I am facing issues after updating my code base. I get the following error whenever I use onboarding_qualification:

```
omegaconf.errors.ConfigKeyError: Key 'onboarding_qualification' not in 'ParlAIChatBlueprintArgs'                                                                                                                                                                                            
full_key: mephisto.blueprint.onboarding_qualification                                                                                                                                                                                                                                   
reference_type=BlueprintArgs                                                                                                                                                                                                                                                            
object_type=ParlAIChatBlueprintArgs
```

I realized that the parameter definition has been removed in the latest commit. See [here](https://github.com/facebookresearch/Mephisto/commit/dfdfc7592c8e401be5cf14e4a60a4ff88cf6217f#r57145702).

Can you please elaborate on this?
